File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for(int i = 0; i < words.length; i++){
            if(isPalindrome(words[i]))
                return words[i];
        }
        return "";
    }
    
    public boolean isPalindrome(String str) {
        int len = str.length();
        
        for(int i = 0; i < len/2; i++){
            if(str.charAt(i) != str.charAt(len-1-i))
                return false;
        }
        return true;
    }
}

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.swift
class Solution {
    func firstPalindrome(_ words: [String]) -> String {
        for str in words {
            if isPalindromic(str) {
                return str
            }
        }
        
        return ""
    }
    
    private func isPalindromic(_ str: String) -> Bool {
        var len = str.count
        var i = 0
        var j = len-1
        
        while i < len/2 {
            let indexI = str.index(str.startIndex, offsetBy: i)
            let indexJ = str.index(str.startIndex, offsetBy: j)
            
            if str[indexI] != str[indexJ] {
                return false
            }
            
            i += 1
            j -= 1
        }

        return true
    }
    
}